util/stream.go

```diff
@@ -32,12 +30,36 @@ type Parser interface {
 	Parse(b []byte) (message Message, err error)
 }
 
+type streamWorker struct {
+	Full chan *bytes.Buffer
+}
+
+func (w *streamWorker) parse(stopCh chan bool, parser Parser, inbound chan Message, empty chan *bytes.Buffer) {
+	errMessage := "received: %v and encountered error: %v"
+	for {
+		select {
+		case b := <-w.Full:
+			msg, err := parser.Parse(b.Bytes())
+			// Log all message parsing errors.
+			if err != nil {
+				log.Errorf(errMessage, b.Bytes(), err)
+			}
+
+			inbound <- msg
+			b.Reset()
+			empty <- b
+		case <-stopCh:
+			return
+		}
+	}
+}
+
 type MessageStream struct {
 	conn net.Conn
 	pool *BufferPool
 	// Message parser
 	parser Parser
-	// Channel to shut down the parser goroutine
+	//Channel to shut down the parser goroutine
```

Comment:
nit: not a necessary change and we typically leave a whitespace at the beginning

Reply:
Done.